src/lib/Microsoft.Health.Fhir.Ingest.Template/JsonPathCalculatedFunctionContentTemplateAdapter.cs

```diff
@@ -16,32 +16,56 @@ public JsonPathCalculatedFunctionContentTemplateAdapter(TTemplate innerTemplate)
+            TypeMatchExpression = CreateExpression(InnerTemplate.TypeMatchExpression, innerTemplate.GetLineInfoForProperty(nameof(InnerTemplate.TypeMatchExpression)));
+            DeviceIdExpression = CreateExpression(InnerTemplate.DeviceIdExpression, innerTemplate.GetLineInfoForProperty(nameof(InnerTemplate.DeviceIdExpression)));
+            PatientIdExpression = CreateExpression(InnerTemplate.PatientIdExpression, innerTemplate.GetLineInfoForProperty(nameof(InnerTemplate.PatientIdExpression)));
+            EncounterIdExpression = CreateExpression(InnerTemplate.EncounterIdExpression, innerTemplate.GetLineInfoForProperty(nameof(InnerTemplate.EncounterIdExpression)));
+            TimestampExpression = CreateExpression(InnerTemplate.TimestampExpression, innerTemplate.GetLineInfoForProperty(nameof(InnerTemplate.TimestampExpression)));
+            CorrelationIdExpression = CreateExpression(InnerTemplate.CorrelationIdExpression, innerTemplate.GetLineInfoForProperty(nameof(InnerTemplate.CorrelationIdExpression)));
 
             if (InnerTemplate.Values != null)
             {
                Values = InnerTemplate.Values.Select(value =>
                new CalculatedFunctionValueExpression()
                {
                    ValueName = value.ValueName,
-                   ValueExpression = new TemplateExpression(value.ValueExpression, TemplateExpressionLanguage.JsonPath),
+                   ValueExpression = CreateExpression(value.ValueExpression, value),
                    Required = value.Required,
+                   LineNumber = value.LineNumber,
+                   LinePosition = value.LinePosition,
+                   LineInfoForProperties = value.LineInfoForProperties,
                }).ToList();
             }
+
+            LineNumber = innerTemplate.LineNumber;
+            LinePosition = innerTemplate.LinePosition;
+            LineInfoForProperties = innerTemplate.LineInfoForProperties;
         }
 
         public TTemplate InnerTemplate { get; private set; }
 
-        private TemplateExpression CreateExpression(string value)
+        private TemplateExpression CreateExpression(string value, ILineInfo lineAwareJsonObject)
         {
             if (!string.IsNullOrWhiteSpace(value))
             {
-                return new TemplateExpression(value, TemplateExpressionLanguage.JsonPath);
+                if (lineAwareJsonObject == null)
+                {
+                    return new TemplateExpression(value, TemplateExpressionLanguage.JsonPath);
+                }
+
+                var templateExpression = new TemplateExpression(value, TemplateExpressionLanguage.JsonPath)
+                {
+                    LineNumber = lineAwareJsonObject.LineNumber,
+                    LinePosition = lineAwareJsonObject.LinePosition,
+                };
+
+                templateExpression.LineInfoForProperties[nameof(TemplateExpression.Value)] = new LineInfo()
```

Comment:
How do we ensure we don't conflict with a property named Value?  Referring to LineAwareJsonObjectExtensions line 45.

Reply:
This is a new `TemplateExpression` object (created on line 56). Its `LineInfoForProperties` object is empty so there would be nothing to conflict with. 

In general, a `TemplateExpression` only has 2 properties: `value` and `language`. Language is optional, so I don't see it its position. Its `value` can be specified 2 ways, shown below. In the first case, we set the position of `value` to that of the overall `TemplateExpression`. In the second case, we are setting its line position to that of the explicit object `value`.

```json
"deviceIdExpression": "$.device"
```

```json
"deviceIdExpression": {
      "value": "$.device"
    },
```
```json
"deviceIdExpression": {
      "value": "$.device",
      "language": "JsonPath"
    },
```

Reply:
Thanks for the clarification.  No concern then!